projects/com.oracle.truffle.llvm.parser.macho.injector/src/com/oracle/truffle/llvm/parser/macho/injector/Injector.java

```diff
@@ -0,0 +1,173 @@
+ * endorse or promote products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.oracle.truffle.llvm.parser.macho.injector;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.oracle.truffle.llvm.parser.macho.MachOFile;
+import com.oracle.truffle.llvm.parser.macho.MachOSegmentCommand;
+import com.oracle.truffle.llvm.parser.macho.MachOSegmentCommand.MachOSection;
+
+public final class Injector {
+
+    public static void main(String[] args) {
+        if (args.length < 1 || args.length > 2) {
+            throw new IllegalArgumentException("Please provide a wllvm input file and optionally an output file!");
+        }
+
+        File file = new File(args[0]);
+        if (!file.exists()) {
+            throw new IllegalArgumentException("File does not exist!" + file.getAbsolutePath());
+        }
+
+        if (args.length == 2) {
+            File output = new File(args[1]);
+            try {
+                file = Files.copy(file.toPath(), output.toPath()).toFile();
+            } catch (IOException e) {
+                throw new RuntimeException("Could not write into specified output file!");
+            }
+        }
+
+        try {
+            injectBitcode(file);
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException("Bitcode could not be injected into WLLVM file!");
+        }
+    }
+
+    public static void injectBitcode(File wllvmFile) throws IOException, InterruptedException {
+
+        File bcFile = Files.createTempFile("extractedBc_", ".bc").toFile();
+        bcFile.deleteOnExit();
+
+        extractBc(wllvmFile, bcFile);
+
+        // read wllvm built Mach-O executable
+        ByteBuffer machOBuffer = readFile(wllvmFile);
+        MachOFile machO = MachOFile.create(machOBuffer);
+
+        MachOSegmentCommand seg = machO.getSegment("__WLLVM");
```

Comment:
Having one (or more) helper methods would increase the readability. For example, you could have a method `getBitcodeSection` that gets the segment and section and performs checks on them.